### ISSUE TITLE
Deprecation warning for `util._extend`

### ISSUE THREAD

**Issue by TheJaredWilcurt** [chimurai/http-proxy-middleware#1017]
### Checks

* [X] I understand project setup issues should be asked on [StackOverflow](https://stackoverflow.com/questions/tagged/http-proxy-middleware) or in [GitHub Discussions](https://github.com/chimurai/http-proxy-middleware/discussions).
* [X] I updated to latest `http-proxy-middleware`.

### Describe the bug (be clear and concise)

In Node v22 they started showing a deprecation warning for usage of `util._extend`, which is used in a dependency of this library.

```
(node:27864) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
    at ProxyServer.<anonymous> (\node_modules\http-proxy\lib\http-proxy\index.js:50:26)
    at HttpProxyMiddleware.middleware (\node_modules\http-proxy-middleware\dist\http-proxy-middleware.js:22:32)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

There is a PR for that dependency:

* https://github.com/http-party/node-http-proxy/pull/1666

Once that is merged and a new version is released, `http-proxy-middleware` will need to update the dependency to stop receiving the deprecation warning in Node v22+.

*I'm mainly making this issue to encourage prioritization of the linked PR.*

### Step-by-step reproduction instructions

```shell
1. Use library with Node 22+
```

### Expected behavior (be clear and concise)

No deprecation warnings

### How is http-proxy-middleware used in your project?

```shell
internal-app@1.0.0 C:\internal-app
`-- @internal-library/utils-ui-dev@3.1.1
  `-- http-proxy-middleware@3.0.0.2 deduped
| +-- istanbul-reports@3.1.7
```

### What http-proxy-middleware configuration are you using?

```js
N/A
```

### What OS/version and node/version are you seeing the problem?

```shell
Node v22.3.0 - All OS
```

### Additional context (optional)

*No response*

**Comment by chimurai** [comment]
Thanks for the report.

With semver, if `http-proxy` releases a new `minor` of `patch` version, there is no need to update this project.

You can just update your package lock file to retrieve updated `http-proxy` dependency.

https://github.com/chimurai/http-proxy-middleware/blob/82aaf0aefaeb341b87fcd5b0e27a15deb030e57e/package.json#L90
---
Closing this issue
(since no update to `http-proxy-middleware` is needed when `http-proxy` releases a minor/patch update)

**Comment by habahabahaba** [comment]
Hello. I am getting "Deprecation warning for util._extend", when using http-proxy-middleware with Express.
My code:
```
// 3rd party:
import { createProxyMiddleware } from 'http-proxy-middleware';
// Express:
import express from 'express';

export function serve(
  port: number,
  filename: string,
  dir: string
): Promise<void> {
  const app = express();

  app.use(
    createProxyMiddleware({
      target: `http://localhost:5173`,
      ws: true,
    })
  );

  return new Promise<void>((resolve, reject) => {
    app.listen(port, resolve).on('error', reject);
  });
}
```
My package.json:
```
{
  "version": "1.0.0",
  "description": "",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "type": "module",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "dev": "tsc --watch --preserveWatchOutput"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@types/cors": "^2.8.17",
    "@types/express": "^4.17.21",
    "cors": "^2.8.5",
    "express": "^4.19.2",
    "http-proxy-middleware": "^3.0.0"
  }
}
```
My error:
```
(node:26057) [DEP0060] DeprecationWarning: The `util._extend` API is deprecated. Please use Object.assign() instead.
    at ProxyServer.<anonymous> (.../node_modules/.pnpm/http-proxy@1.18.1_debug@4.3.5/node_modules/http-proxy/lib/http-proxy/index.js:50:26)
    at HttpProxyMiddleware.middleware (.../node_modules/.pnpm/http-proxy-middleware@3.0.0/node_modules/http-proxy-middleware/dist/http-proxy-middleware.js:22:32)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by btkfrank** [comment]
> Thanks for the report.
> 
> With semver, when `http-proxy` releases a new `minor` of `patch` version, there is no need to update this project.
> 
> You can just update your package lock file to retrieve updated `http-proxy` dependency.
> 
> ##
> https://github.com/chimurai/http-proxy-middleware/blob/82aaf0aefaeb341b87fcd5b0e27a15deb030e57e/package.json#L90
> 
> Closing this issue (since no update to `http-proxy-middleware` is needed when `http-proxy` releases a minor/patch update)
> 
> You can read more about semver: https://semver.org

http-proxy@1.18.1 is already the latest version, and hasn't been updated for 4 years. [And is still using `util._extend`](https://github.com/http-party/node-http-proxy/blob/master/lib/http-proxy/index.js#L50)

**Comment by Netail** [comment]
@chimurai please reopen this issue again. Facing the same deprecation warning with the lastest version :/

**Comment by chabb** [comment]
You can take a look at vitejs. They patched the http-proxy package to solve this issue

**Comment by baur** [comment]
same problem

**Comment by Auios** [comment]
Getting this issue.

**Comment by cami-dev** [comment]
Why was this closed? This is still an issue. You have a dead dependency then called http-proxy which haven't been updated in 5 years. So there wont be any update there. Please replace your dependency to them and fix this issue. Getting this since angular uses http-proxy-middleware as a dependency so can't really be fixed elsewhere. Please reopen. 

Having the same issue with node 22.11.0

**Comment by cami-dev** [comment]
There seem to be something in the work here https://github.com/http-party/node-http-proxy/pull/1666  at http-proxy but ye still no release for those tracking this issue .

**Comment by Netail** [comment]
Yeah I doubt there will be an updating coming in node-http-proxy any time soon... Best to switch to a fork, write a patch or switch a proxy implementation dependency

**Comment by cami-dev** [comment]
@chimurai see above comments please.

**Comment by uvbeenzaned** [comment]
Those that need a temporary fix can use a commit from a [PR that was submitted to the http-proxy package](https://github.com/http-party/node-http-proxy/pull/1666) to get rid of the issue until its officially merged. Using overrides in package.json, you can accomplish this. Probably best check back in every so often to see if it gets merged into the main project.

```json
"overrides": {
    "http-proxy-middleware": {
      "http-proxy": "github:nikeee/node-http-proxy#b42c38fc3486ee8ad4374cc2397458b6e4cd99ca"
    }
}
```

**Comment by chimurai** [comment]
Thanks for reviving `http-proxy` with `http-proxy-3` @williamstein

When the generics implement, I'll definitely consider to migrate to `http-proxy-3`

Currently, I also don't have time to implement it. Hopefully the community can help out to make this happen  🙏
